app/src/main/java/lt/getpet/getpet/PetProfileActivity.kt

```diff
@@ -78,16 +81,30 @@ class PetProfileActivity : BaseActivity() {
 
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         if (item.itemId == R.id.delete_pet) {
-            val disposable = petsService.savePetChoice(pet, false)
-                    .subscribeOn(ioScheduler)
-                    .observeOn(uiScheduler)
-                    .subscribe({
-                        finish()
-                    }, {
-                        Timber.w(it)
-                    })
-
-            subscriptions.add(disposable)
+            val builder = AlertDialog.Builder(this, R.style.MyDialogTheme)
+            builder.setTitle(R.string.dialog_delete_title)
+                    .setMessage(R.string.dialog_delete_message)
+                    .setPositiveButton(R.string.button_ok
+                    ) { dialog, id ->
+                        val disposable = petsService.savePetChoice(pet, false)
+                                .subscribeOn(ioScheduler)
+                                .observeOn(uiScheduler)
+                                .subscribe({
+                                    finish()
+                                }, {
+                                    Timber.w(it)
+                                })
+                        subscriptions.add(disposable)
+                    }
+                    .setNegativeButton(R.string.button_cancel
+                    ) { dialog, id ->
+                        dialog.cancel() // User cancelled the dialog
+                    }
+            // Create the AlertDialog object and return it
+            builder.create()
```

Comment:
The is no need to call `create` as you don't need `builder` anywhere. Show will build dialog automatically. 
From `show` documentation:
```
Creates an {@link AlertDialog} with the arguments supplied to this
builder and immediately displays the dialog.
Calling this method is functionally identical to:
AlertDialog dialog = builder.create();
dialog.show();
```

Reply:
Thank you for the tip. Code shortened :) :+1: